Dockerfile

```diff
@@ -8,26 +8,23 @@ ARG GOLANG_IMAGE=docker.io/library/golang:1.24.1@sha256:c5adecdb7b3f8c5ca3c88648
 # Represents the plataform where the build is happening, do not mix with
 # TARGETARCH
 FROM --platform=${BUILDPLATFORM} ${GOLANG_IMAGE} AS builder
+WORKDIR /go/src/github.com/cilium/cilium-cli
 
 # TARGETOS is an automatic platform ARG enabled by Docker BuildKit.
 ARG TARGETOS
 # TARGETARCH is an automatic platform ARG enabled by Docker BuildKit.
 ARG TARGETARCH
-
-WORKDIR /go/src/github.com/cilium/cilium-cli
-
 RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium-cli \
     --mount=type=cache,target=/root/.cache \
     --mount=type=cache,target=/go/pkg \
     make GOARCH=${TARGETARCH} DESTDIR=/out/${TARGETOS}/${TARGETARCH} install
 
 FROM gcr.io/distroless/static:latest@sha256:95ea148e8e9edd11cc7f639dc11825f38af86a14e5c7361753c741ceadef2167 AS release
+LABEL maintainer="maintainer@cilium.io"
+ENTRYPOINT []
+WORKDIR /root/app
```

Comment:
As these layers are least likely to change we create them first, this means during builds, pulls, and push of the image there are less layers to be touched as they can be reused.

The first layer that changes, means all following layers have to be rebuild.